plugin-viewport/src/main/kotlin/com/mapbox/maps/plugin/viewport/experimental/ViewportPluginImpl.kt

```diff
@@ -0,0 +1,66 @@
+package com.mapbox.maps.plugin.viewport.experimental
+
+import com.mapbox.maps.plugin.viewport.experimental.state.ViewportState
+import com.mapbox.maps.plugin.viewport.experimental.state.ViewportStateChangedObserver
+import com.mapbox.maps.plugin.viewport.experimental.transition.TransitionEndListener
+import com.mapbox.maps.plugin.viewport.experimental.transition.ViewportTransitionOptions
+import java.util.concurrent.CopyOnWriteArrayList
+
+/**
+ * Mapbox default implementation for [ViewportPlugin].
+ */
+class ViewportPluginImpl : ViewportPlugin {
+  private val registeredViewportStates = CopyOnWriteArrayList<ViewportState>()
+  private var currentState: ViewportState = ViewportState.IDLE
+
+  override val state: ViewportState
```

Comment:
Should it be nullable? What if you…
* haven't registered any states
* start interacting with the map and are no longer in any state

Reply:
I guess we could represent IDLE with null currentState

Reply:
Renamed it to currentState and made it nullable to represent IDLE state.